### ISSUE TITLE
Need HasValue at editor time, why is it private?

### ISSUE THREAD

**Issue by MeisterDerMagie** [starikcetin/Eflatun.SceneReference#70]
The title says it all. I need to check if a scene reference has assigned a scene or not, to throw a warning if it's missing. Is there any way to achieve this?
Using State.Unsafe doesn't work because it will mark a scene that's referenced but not in the build settings as Unsafe. I specifically need to know if the designers referenced a scene.

**Comment by starikcetin** [comment]
It was marked private because the validation API was getting out of hand. There was a lot of partial validations with a lot of possibilities, which was confusing even me as the original developer. It made sense to pack them into a simplified version.

While I see your point of view, I believe what you need is not necessarily accessing `HasValue`, but rather a mechanism for us to report why the reference was deemed unsafe. Something along the lines of this:

```cs
enum UnsafeReason {
  Empty,
  NotInBuild,
  // ...etc
}
```

And then:

```cs
if(_sceneRef.UnsafeReason === UnsafeReason.Empty) {
}
```

Would something that work for you? What do you think?

**Comment by MeisterDerMagie** [comment]
That sounds like a good solution because generally gives more information if needed. Only thing is, how would you implement it? Unsafe is an enum itself, you can't return any additional values.

**Comment by starikcetin** [comment]
> That sounds like a good solution because it generally provides more information if needed. Only thing is, how would you implement it? Unsafe is an enum itself, you can't return any additional values.

It would be a separate enum provided via a separate property called `UnsafeReason`. It can also contain a `None` entry to prevent querying both `State` and `UnsafeReason` all the same time.

**Comment by starikcetin** [comment]
Implemented in #80. Will be included in the next release.

**Comment by starikcetin** [comment]
Included in [3.2.0](https://github.com/starikcetin/Eflatun.SceneReference/releases/tag/3.2.0).
